src/yaml.rs

```diff
@@ -199,6 +212,62 @@ impl YamlLoader {
     }
 }
 
+pub struct YamlDecoder<T: std::io::Read> {
+  source: T,
+  trap: encoding::types::DecoderTrap,
+}
+
+
+impl<T: std::io::Read> YamlDecoder<T> {
+  pub fn read(source: T) -> YamlDecoder<T> {
+    YamlDecoder{
+      source: source,
+      trap: encoding::DecoderTrap::Strict,
+    }
+  }
+
+  pub fn encoding_trap(mut self, trap: encoding::types::DecoderTrap) -> YamlDecoder<T> {
```

Comment:
@mkmir @17cupsofcoffee suggested something like the following might be better for a builder:
```suggestion
  pub fn encoding_trap(&mut self, trap: encoding::types::DecoderTrap) -> &mut Self {
```

Reply:
Thanks; PTAL

Reply:
Looks reasonable, thanks. I think it's now up to the project maintainers to decide whether they like the style.